dom-weather-app/public/app.js

```diff
@@ -0,0 +1,92 @@
+const cityInput = document.querySelector('#city-input');
+const startButton = document.querySelector('#startButton');
+
+const apiKey = 'kUfkPA9VVU5407oDJWDoh2T10ZALToRm';
+
+const allWeatherInformation = async () => {
+    const cityDetails = await getCity(cityInput.value);
+    const weatherConditions = await getWeather(cityDetails.cityID);
+
+    return {
+        isDay: weatherConditions[0].IsDaylight,
+        temperatureInFahrenheits: weatherConditions[0].Temperature.Value,
+        iconPhrase: weatherConditions[0].IconPhrase,
+        weatherIcon: weatherConditions[0].WeatherIcon
+    }
+}
+
+startButton.addEventListener('click', async () => {
+    updateUI(
+        (await allWeatherInformation()).isDay,
+        (await allWeatherInformation()).temperatureInFahrenheits,
+        (await allWeatherInformation()).iconPhrase,
+        (await allWeatherInformation()).weatherIcon
+    );
+})
+
+
+async function getCity (inputValue) {
+
+    const base = 'https://dataservice.accuweather.com/locations/v1/cities/search';
+    const query = `?apikey=${apiKey}&q=${inputValue}`;
+
+    const response = await fetch(base + query)
+        .catch(err => {
+            console.error(err);
+        })
+    const data = await response.json();
+
+    return {cityID: data[0].Key};
+
+}
+
+async function getWeather (cityKey) {
+    const base = 'https://dataservice.accuweather.com/forecasts/v1/hourly/1hour/';
+    const query = `${cityKey}?apikey=${apiKey}`;
+
+    const response = await fetch(base + query)
+        .catch(err => {
+            console.error(err);
+        });
+    const weatherData = await response.json();
+
+    return weatherData;
+}
+
+function updateUI(isDay, temperatureInFahrenheits, iconPhrase, weatherIcon) {
+    changeBackground(isDay);
+    updateTemperature(fahrenheitToCelsius(temperatureInFahrenheits));
+    updateWeatherConditionsAndIcons(iconPhrase, weatherIcon);
+}
+
+function changeBackground (isDayTime) {
```

Comment:
to też w sumie 2 funkcja, jedna robi changebackground i dostaje np tylko SVG string, a inna robi determine

Reply:
hmm chociaż tak naprawde, mi z API przychodzi już true / false => więc ta funkcja nic by nie robiła tylko powielała to co już dostaje :thinking: